aspnetcore/blazor/globalization-localization.md

```diff
@@ -97,63 +220,271 @@ By default, the Intermediate Language (IL) Linker configuration for Blazor WebAs
+
+Inside the closing `</body>` tag and after the Blazor script of `wwwroot/index.html`, provide JS functions to get and set the user's culture selection with browser local storage:
 
-    ...
-</body>
+```html
+<script>
+  window.blazorCulture = {
+    get: () => window.localStorage['BlazorCulture'],
+    set: (value) => window.localStorage['BlazorCulture'] = value
+  };
+</script>
+```
+
+::: moniker range=">= aspnetcore-5.0"
+
+> [!NOTE]
+> The preceding example pollutes the client with global methods. For a better approach in production apps, see [JavaScript isolation in JavaScript modules](xref:blazor/js-interop/call-javascript-from-dotnet#javascript-isolation-in-javascript-modules).
+>
+> Example:
+>
+> ```javascript
+> export function getBlazorCulture() {
+>   return window.localStorage['BlazorCulture'];
+> };
+> export function setBlazorCulture(value) {
+>   window.localStorage['BlazorCulture'] = value;
+> };
+> ```
+>
+> If you use the preceding functions, change the JS interop calls in this section from `blazorCulture.get` to `getBlazorCulture` and from `blazorCulture.set` to `setBlazorCulture`.
+
+::: moniker-end
+
+In `Program.Main` (`Program.cs`):
+
+* Add namespaces for <xref:System.Globalization?displayProperty=fullName> and <xref:Microsoft.JSInterop?displayProperty=fullName>.
+* Add Blazor's localization service to the app's service collection with <xref:Microsoft.Extensions.DependencyInjection.LocalizationServiceCollectionExtensions.AddLocalization%2A>.
+* Use [JS interop](xref:blazor/js-interop/call-javascript-from-dotnet) to call into JS and retrieve the user's culture selection from local storage. If local storage doesn't contain a culture for the user, set a default value.
+
+Make the following adjustments to `Program.Main`:
+
+```diff
+using System;
+using System.Net.Http;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Text;
+using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
++using System.Globalization;
++using Microsoft.JSInterop;
+
+namespace SampleApp
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var builder = WebAssemblyHostBuilder.CreateDefault(args);
+            builder.RootComponents.Add<App>("#app");
+
+            builder.Services.AddScoped(
+                sp => new HttpClient 
+                { 
+                    BaseAddress = new Uri(builder.HostEnvironment.BaseAddress)
+                });
+
+-           await builder.Build().RunAsync();
++           builder.Services.AddLocalization();
+
++           var host = builder.Build();
+
++           CultureInfo culture;
++           var js = host.Services.GetRequiredService<IJSRuntime>();
++           var result = await js.InvokeAsync<string>("blazorCulture.get");
```

Comment:
This is what I mean to refer to in my previous comment about shows how the javascript function is called